instance.tf

```diff
@@ -1,5 +1,5 @@
 resource "aws_instance" "public" {
-  ami                         = "ami-06e85d4c3149db26a"
+  ami                         = data.aws_ami.amazonlinux.id
```

Comment:
reference to the aws_ami datasource was made but it wasn't declared anywhere in the repository. 
could you please declare the aws_ami datasource in a new file `data.tf`